client/tests/e2e.py

```diff
@@ -0,0 +1,181 @@
+    json_ret = req.json()
+    assert (json_ret == users)
+
+
+def test_providers():
+    req = requests.get("http://localhost:7000/data/providers", json=True)
+    json_ret = req.json()
+    assert (json_ret == providers)
+
+
+def test_sources():
+    retries = 0
+    all_ready = False
+    json_ret = None
+    while 1:
+        if all_ready:
+            break
+        req = requests.get("http://localhost:7000/data/sources", json=True)
+        json_ret = req.json()
+        if retries > 20:
+            assert (json_ret == sources)
+        all_ready = True
+        for res in json_ret:
+            for v in res['variants']:
+                del res['variants'][v]['created']
+                if res['variants'][v]['status'] != 'READY':
+                    time.sleep(5)
+                    retries += 1
+                    all_ready = False
+                    break
+
+    assert (json_ret == sources)
+
+
+def test_entities():
+    req = requests.get("http://localhost:7000/data/entities", json=True)
+    json_ret = req.json()
+    assert (json_ret == entities)
+
+
+def test_features():
+    retries = 0
+    all_ready = False
+    json_ret = None
+    while 1:
+        if all_ready:
+            break
+        req = requests.get("http://localhost:7000/data/features", json=True)
+        json_ret = req.json()
+        if retries > 20:
+            assert (json_ret == features)
+        all_ready = True
+        for res in json_ret:
+            for v in res['variants']:
+                del res['variants'][v]['created']
+                if res['variants'][v]['status'] != 'READY':
+                    time.sleep(5)
+                    retries += 1
+                    all_ready = False
+                    break
+
+    assert (json_ret == features)
+
+
+
+def test_labels():
+    retries = 0
+    all_ready = False
+    json_ret = None
+    while 1:
+        if all_ready:
+            break
+        req = requests.get("http://localhost:7000/data/labels", json=True)
+        json_ret = req.json()
+        if retries > 20:
+            assert (json_ret == labels)
+        all_ready = True
+        for res in json_ret:
+            for v in res['variants']:
+                del res['variants'][v]['created']
+                if res['variants'][v]['status'] != 'READY':
+                    time.sleep(5)
+                    retries += 1
+                    all_ready = False
+                    break
+
+    assert (json_ret == labels)
+
+
+def test_training_sets():
+    retries = 0
+    all_ready = False
+    json_ret = None
+    while 1:
+        if all_ready:
+            break
+        req = requests.get("http://localhost:7000/data/training-sets", json=True)
+        json_ret = req.json()
+        if retries > 20:
+            assert (json_ret == training_sets)
+        all_ready = True
+        for res in json_ret:
+            for v in res['variants']:
+                del res['variants'][v]['created']
```

Comment:
Is there a reason for this delete?

Reply:
Yeah everytime a resource is created it gets a timestamp. Theres not a great way to test the exact time the resource was created, since it depends on a lot of external sources, so I remove it instead